tests/dummy.c

```diff
@@ -20,6 +20,13 @@ enum bpf_map_type {
 	BPF_MAP_TYPE_CGROUP_ARRAY,
 };
 
+enum bpf_pin_type {
+    PIN_NONE = 0,
+    PIN_OBJECT_NS,
+    PIN_GLOBAL_NS,
+    PIN_CUSTOM_NS,
+};
```

Comment:
We should have a canonical header file that contains the definition of `bpf_pin_type` and `bpf_map_def` instead of letting projects using gobpf guessing. Then, this test file can `#include` it.

At the moment, `struct bpf_map_def` is defined in `elf/elf.go` and `tests/dummy.c` and the definitions don't match betcause it's difficult to keep things up-to-date with copy/paste.

Reply:
@alban Updated. Now common structures and types are defined in `elf/include/bpf.h`.